src/processes/follow-topic-process.js

```diff
@@ -1,7 +1,95 @@
 const StreamChat = require('stream-chat').StreamChat;
+const crypto = require('crypto');
+const {
+  LogError,
+  UserTopic,
+  User,
+  CommunityMessageFormat,
+  ChatAnonUserInfo
+} = require('../databases/models');
+const BetterSocialConstantListUtils = require('../utils/constantList/utils');
+const CHANNEL_TYPE_ANONYMOUS = require('../utils/constant');
+const updateBetterChannelMembers = require('../getstream/updateBetterChannelMembers');
 
-const {LogError, UserTopic, CommunityMessageFormat} = require('../databases/models');
+const generate_channel_id_for_anon_chat = (owner, member, context = null, source_id = null) => {
+  const hash = crypto.createHash('sha256');
+  hash.update(owner);
+  hash.update(member);
+  if (context) hash.update(context);
+  if (source_id) hash.update(source_id);
 
+  return hash.digest('hex');
+};
+const anon_to_sign_user = async (client, user, targetUser) => {
+  const channel = await ChatAnonUserInfo.findOne({
+    where: {
+      my_anon_user_id: user,
+      target_user_id: targetUser,
+      context: 'auto message',
+      initiator: user
+    }
+  });
+  if (channel) {
+    return channel;
+  } else {
+    const channel_id = generate_channel_id_for_anon_chat(user, targetUser, 'autoMessage');
+    const emoji = BetterSocialConstantListUtils.getRandomEmoji();
+    const color = BetterSocialConstantListUtils.getRandomColor();
+    const anon_init_data = {
+      anon_user_info_color_code: color.code,
+      anon_user_info_color_name: color.color,
+      anon_user_info_emoji_code: emoji.emoji,
+      anon_user_info_emoji_name: emoji.name
+    };
+
+    const new_channel = await ChatAnonUserInfo.create({
+      channel_id: channel_id,
+      my_anon_user_id: user,
+      target_user_id: targetUser,
+      anon_user_info_color_code: anon_init_data.anon_user_info_color_code,
+      anon_user_info_color_name: anon_init_data.anon_user_info_color_name,
+      anon_user_info_emoji_code: anon_init_data.anon_user_info_emoji_code,
+      anon_user_info_emoji_name: anon_init_data.anon_user_info_emoji_name,
+      context: 'auto message',
+      initiator: user,
+      source_id: ''
+    });
+    return new_channel;
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `anon_to_sign_user` function, you are creating a new anonymous user if one doesn't already exist. This seems fine, but there is no error handling for the case where the `ChatAnonUserInfo.create` operation fails. It would be good to wrap this in a try-catch block and handle any potential errors that might occur during the creation of a new anonymous user.

```diff
    try {
      const new_channel = await ChatAnonUserInfo.create({
        channel_id: channel_id,
        my_anon_user_id: user,
        target_user_id: targetUser,
        anon_user_info_color_code: anon_init_data.anon_user_info_color_code,
        anon_user_info_color_name: anon_init_data.anon_user_info_color_name,
        anon_user_info_emoji_code: anon_init_data.anon_user_info_emoji_code,
        anon_user_info_emoji_name: anon_init_data.anon_user_info_emoji_name,
        context: 'auto message',
        initiator: user,
        source_id: ''
      });
      return new_channel;
    } catch (error) {
      console.error('Error creating new anonymous user:', error);
      throw error; // or handle the error appropriately
    }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->